ang/civicase/CaseDetailsPeople.js

```diff
@@ -313,7 +313,46 @@
         // Reset if out of range
         $scope.rolesPage = 1;
       }
+
       $scope.roles = _.slice(caseRoles, (25 * ($scope.rolesPage - 1)), 25 * $scope.rolesPage);
+      setStartDateToClients(item.case_id);
+    }
+
+    /**
+     * Function to get the start date of the contacts with the
+     * Client roles
+     *
+     * @params {Integer} case_id
+     */
+    function setStartDateToClients (caseId) {
+      var params = {
+        'sequential': 1,
+        'return': ['activity_type_id', 'activity_date_time', 'target_contact_id'],
+        'case_id': caseId,
+        'options': {
+          sort: 'activity_type_id DESC'
+        },
+        'activity_type_id': {
+          IN: ['Add Client To Case', 'Reassigned Case']
+        }
+      };
+
+      crmApi('Activity', 'get', params).then(function (res) {
```

Comment:
@swastikpareek please add unit test for this functionality as this is completely new logic.